commands/serve.js

```diff
@@ -211,22 +211,28 @@ function devServerConfig(host, port, protocol, publicPath, proxy, allowedHost) {
 		},
 		// `proxy` is run between `before` and `after` `webpack-dev-server` hooks
 		proxy,
-		onBeforeSetupMiddleware(devServer) {
-			// Keep `evalSourceMapMiddleware`
-			// middlewares before `redirectServedPath` otherwise will not have any effect
-			// This lets us fetch source contents from webpack for the error overlay
-			devServer.app.use(evalSourceMapMiddleware(devServer));
+		setupMiddlewares(middlewares, devServer) {
+			if (!devServer) {
+				throw new Error('webpack-dev-server is not defined');
+			}
 
 			// Optionally register app-side proxy middleware if it exists
 			const proxySetup = path.join(process.cwd(), 'src', 'setupProxy.js');
```

Comment:
Is it safe to leave this here?

Reply:
We can leave the `proxySetup` code in `setupMiddleware()`: it works the same as the previous one(`onBeforeSetupMiddleware()`) and it works the same as the CRA